drivers/i2s/i2s_ll_stm32.c

```diff
@@ -223,6 +223,13 @@ static int i2s_stm32_configure(struct device *dev, enum i2s_dir dir,
 		return ret;
 	}
 
+	/* set I2S Master Clock */
+	if (stream->master) {
+		LL_I2S_EnableMasterClock(cfg->i2s);
+	} else {
+		LL_I2S_DisableMasterClock(cfg->i2s);
+	}
+
```

Comment:
LGTM
I have always used the serial clock (CK).
If MCK has to be used mast be enabled.